src/config/configLoader.ts

```diff
@@ -0,0 +1,73 @@
+import { existsSync, readFileSync, readJSONSync } from 'fs-extra';
+import * as yaml from 'js-yaml';
+import { merge } from 'lodash';
+import * as path from 'path';
+import { ConfigFormat, IConfig } from '../types/config';
+
+const DEFAULT_CONFIG_FILE = 'spectral.yml';
+
+const loadJsonFile = (filePath: string) => {
+  if (existsSync(filePath)) {
+    return readJSONSync(filePath);
+  }
+  throw new Error(`${filePath} does not exist`);
+};
+
+const loadYamlFile = (filePath: string) => {
+  if (existsSync(filePath)) {
+    return yaml.safeLoad(readFileSync(filePath, 'utf8')) || {};
+  }
+  throw new Error(`${filePath} does not exist`);
+};
+
+const loadConfig = (filePath: any): IConfig => {
+  let config;
+
+  switch (path.extname(filePath)) {
```

Comment:
As stated earlier, we can utilize `parseWithPointers` function (it handles JSON just fine).

```ts
import { parseWithPointers } from '@stoplight/yaml';

const loadConfig = (filePath: string): IConfig => {
  try {
    const { data } = parseWithPointers(readFileSync(filePath, 'utf8'));
    // validate data if needed and throw if it's entirely iirrelevant
    return data;
   } catch (ex) {
     // rethrow some common error here, like InvalidConfig or whatever is already defined.
   }
}
```

Reply:
Are you still planning to address this, ya'all?
@mallachari @philsturgeon

Reply:
Sure lets do that.